.github/actions/run-tests/action.yml

```diff
@@ -0,0 +1,77 @@
+name: 'run-tests'
+description: 'Runs go test and upload a step summary'
+inputs:
+  filter:
+    description: 'The go test pattern for the tests to run'
+    required: false
+    default: ''
+  upload-logs-name:
+    description: 'Choose the name of the log artifact'
+    required: false
+    default: logs-${{ github.job }}-${{ strategy.job-index }}
```

Comment:
I am not too sure the `strategy` object exists in a composite action. quiet a few objects don't exists in a composite action which why we need to pass things using inputs.

I'll suggest to use the run number and the run attemps to make it unique `github.run_id` and `github.run_attemps`: https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

Reply:
my bad, should be fine then.

thank you for the improvement, it looks awesome on the summary page.